Sources/Build/Buildable.swift

```diff
@@ -9,20 +9,43 @@
 */
 
 import PackageType
+import struct Utility.Path
 
 protocol Buildable {
     var targetName: String { get }
     var isTest: Bool { get }
 }
 
+extension CModule {
+    func workingDirectory(prefix: String) -> String {
+        return Path.join(prefix, "\(c99name).build")
+    }
+}
+
 extension Module: Buildable {
     var isTest: Bool {
         return self is TestModule
     }
 
-    var Xcc: [String] {
+    func XccFlagsForPrefix(prefix: String) -> [String] {
         return recursiveDependencies.flatMap { module -> [String] in
-            if let module = module as? CModule {
+            if let module = module as? ClangModule {
+                var moduleMap: String? = nil
+                
+                if module.moduleMapPath.isFile {
+                    moduleMap = module.moduleMapPath
+                }
+    
+                let genModuleMap = Path.join(module.workingDirectory(prefix), module.moduleMap)
+                if genModuleMap.isFile {
+                    moduleMap = genModuleMap
+                }
```

Comment:
Does it require copying all the headers? We have to use absolute paths anyway, and then we can just `-I` the package directory.